src/phoenix/auth.py

```diff
@@ -33,16 +61,165 @@ def validate_password_format(password: str) -> None:
     """
     Checks that the password has a valid format.
     """
-    if not password:
-        raise ValueError("Password must be non-empty")
-    if any(char.isspace() for char in password):
-        raise ValueError("Password cannot contain whitespace characters")
-    if not password.isascii():
-        raise ValueError("Password can contain only ASCII characters")
-    if not len(password) >= MIN_PASSWORD_LENGTH:
-        raise ValueError(f"Password must be at least {MIN_PASSWORD_LENGTH} characters long")
+    PASSWORD_REQUIREMENTS.validate(password)
+
+
+@dataclass(frozen=True)
+class _PasswordRequirements:
```

Comment:
This seems unrelated to key authentication - next time can you split so it's easier to review in isolation and also so we can parallelize work a bit better

Reply:
yes, it's not necessary. It's just intended for completeness by smoothing out the password  generation step in the integration test.

Reply:
sg